File: 2984486(small)/jmteric/5752104073297920/0/extracted/ProperShuffle.scala
import java.io.PrintWriter
import scala.io.Source

object ProperShuffle extends App {

  val name = "C-small-attempt2.in"
  val out = new PrintWriter(s"target/$name.out.txt")
  val lines = Source.fromInputStream(getClass.getResourceAsStream(name)).getLines()

  val caseCount = lines.next().toInt

  for (n <- 1 to caseCount) {
    lines.next() //ignore
    var data = lines.next().split(' ').map(_.toInt)

    val sampleSize = data.size / 3.0

    val lowIndexes = for ((x, i) <- data.zipWithIndex if x < sampleSize) yield i
    val lowAvg = lowIndexes.sum / lowIndexes.size.toDouble


    val highIndexes = for ((x, i) <- data.zipWithIndex if x >= (data.length - sampleSize)) yield i
    val highAvg = highIndexes.sum / highIndexes.size.toDouble

    val gap = (highAvg - lowAvg) / data.size

    println(gap)

    val result = if (gap > .02) "BAD" else "GOOD"

    out.println(s"Case #$n: $result")
  }
  out.close()

}

File: 2984486(small)/stolis/5752104073297920/0/extracted/ProperShuffle.java
package round1;

import java.io.File;
import java.io.PrintWriter;
import java.util.Scanner;

public class ProperShuffle {
    
    public static void main(String[] args) throws Exception {
        File inputFile = new File("C-small-attempt0.in");
        Scanner in = new Scanner(inputFile);
        File outputFile = new File("output.txt");
        PrintWriter out = new PrintWriter(outputFile);

        int T = in.nextInt();
        for (int t=0; t<T; t++) {
            int N = in.nextInt();
            long sum = 0;
            for (int n=0; n<N; n++) {
                int an = in.nextInt();                
                if (an != 0 && n != N-1) {
                    sum += an-(n+1);
                    if (an < n) {
                        sum += N;
                    }
                }
            }
            boolean good = (497540-sum < sum-469145);
            
            out.println("Case #"+(t+1)+": "+(good ? "GOOD" : "BAD"));
        }

        out.close();
    }
    
}
